src/Symfony/Component/Cache/Traits/FilesystemTrait.php

```diff
@@ -23,6 +23,23 @@
     use FilesystemCommonTrait;
 
     /**
+     * @return bool
+     */
+    public function prune()
+    {
+        $time = time();
+        $pruned = true;
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->directory, \FilesystemIterator::SKIP_DOTS), \RecursiveIteratorIterator::LEAVES_ONLY, \RecursiveIteratorIterator::CATCH_GET_CHILD) as $file) {
+            if ($time >= $file->getMTime()) {
```

Comment:
I think that `mtime` is not reliable enough for this use, especially on Windows or other "strange" filesystems that might be used for a file cache.

The current `FilesystemCache` implementation writes the expiry time to the first line of the file to avoid needing to use `mtime`, for this reason.